app/screens/browse_channels/browse_channels.tsx

```diff
@@ -0,0 +1,501 @@
+        doGetChannels();
+
+        // Since doGetChannels also depends on typeOfChannels, it should be up to date when reaching this effect.
+    }, [typeOfChannels]);
+
+    useDidUpdate(() => {
+        const active = selectActiveChannels();
+        if (term) {
+            setVisibleChannels(filterChannelsByTerm(active, term));
+        } else {
+            setVisibleChannels(active);
+        }
+    }, [selectActiveChannels]);
+
+    useEffect(() => {
+        // Update header buttons in case anything related to the header changes
+        setHeaderButtons(!adding);
+    }, [theme, canCreateChannels]);
+
+    let content;
+    if (adding) {
+        content = renderLoading();
+    } else {
+        let channelDropdown;
+        if (canShowArchivedChannels || sharedChannelsEnabled) {
+            channelDropdown = (
+                <ChannelDropdown
+                    onPress={onPressDropdownElement}
+                    typeOfChannels={typeOfChannels}
+                    canShowArchivedChannels={canShowArchivedChannels}
+                    sharedChannelsEnabled={sharedChannelsEnabled}
+                />
+            );
+        }
+
+        content = (
+            <>
+                <View
+                    testID='more_channels.screen'
+                    style={style.searchBar}
+                >
+                    <SearchBar
+                        testID='more_channels.search_bar'
+                        placeholder={intl.formatMessage({id: 'search_bar.search', defaultMessage: 'Search'})}
+                        cancelTitle={intl.formatMessage({id: 'mobile.post.cancel', defaultMessage: 'Cancel'})}
+                        backgroundColor='transparent'
+                        inputHeight={33}
+                        placeholderTextColor={changeOpacity(theme.centerChannelColor, 0.5)}
+                        tintColorSearch={changeOpacity(theme.centerChannelColor, 0.5)}
+                        tintColorDelete={changeOpacity(theme.centerChannelColor, 0.5)}
+                        titleCancelColor={theme.centerChannelColor}
+                        inputStyle={style.searchBarInput}
+                        onChangeText={searchChannels}
+                        onSearchButtonPress={searchChannels}
+                        onCancelButtonPress={stopSearch}
+                        autoCapitalize='none'
+                        keyboardAppearance={getKeyboardAppearanceFromTheme(theme)}
+                        value={term}
+                    />
+                </View>
+                {channelDropdown}
+                <FlatList
+                    data={visibleChannels}
+                    renderItem={renderItem}
+                    testID='more_channels.flat_list'
+                    ListEmptyComponent={loading ? renderLoading : renderNoResults}
+                    onEndReached={doGetChannels}
+                    ListFooterComponent={loading && visibleChannels.length ? renderLoading : null}
```

Comment:
If we don't filter out the case when there are no visible channels, we will render twice the loading, one for the footer component and another one for the list empty component.